components/events.jsx

```diff
@@ -37,10 +57,76 @@ export default () => {
           <tbody>
             {events.map((event) => (
               <tr key={event.uri}>
-                <td><Link to={`/events/${event.uri.split('/')[4]}`}>{event.name}</Link></td>
-                <td>{event.date}</td>
-                <td>{event.start_time}</td>
-                <td>{event.end_time}</td>
+                {selectedOption === 'all-events' && (
+                  <React.Fragment>
+                    <td>
+                      <Link to={`/events/${event.uri.split('/')[4]}`}>
+                        {event.name}
+                      </Link>
+                    </td>
+                    <td>{event.date}</td>
+                    <td>{event.start_time}</td>
+                    <td>{event.end_time}</td>
+                  </React.Fragment>
+                )}
+
+                {selectedOption === 'active-events' &&
```

Comment:
Ideally rather than doing this client-side when the user selects an event filter we should make another server-side request and pass the relevant query params to Calendly API. As you've noticed when we do this client-side it sort of breaks pagination. I'd imagine when an option is selected we'd call `fetchData` again and pass through a param to know what query param we need to pass to Calendly API.

Reply:
Ah, I agree! I'll do that route. More to come. Thank you!